Tests/Realm.Tests/Database/CollectionTests.cs

```diff
@@ -411,6 +412,317 @@ public void Results_GetFiltered_SanityTest()
             Assert.That(query.ToArray().All(i => i.Int >= 5));
         }
 
+        public struct StringQueryNumericData
+        {
+            public string PropertyName;
+            public RealmValue ValueToAddToDB;
+            public RealmValue ValueToQueryFor;
+            public bool ExpectedMatch;
+
+            public StringQueryNumericData(string propertyName, RealmValue valueToAddToDB, RealmValue valueToQueryFor, bool expectedMatch)
+            {
+                PropertyName = propertyName;
+                ValueToAddToDB = valueToAddToDB;
+                ValueToQueryFor = valueToQueryFor;
+                ExpectedMatch = expectedMatch;
+            }
+        }
+
+        public struct StringQueryTestData
+        {
+            public string PropertyName;
+            public RealmValue MatchingValue;
+            public RealmValue NonMatchingValue;
+
+            public StringQueryTestData(string propertyName, RealmValue matchingValue, RealmValue nonMatchingValue)
+            {
+                PropertyName = propertyName;
+                MatchingValue = matchingValue;
+                NonMatchingValue = nonMatchingValue;
+            }
+
+        }
+
+        public static object BoxValue(RealmValue val, Type targetType)
+        {
+            var boxed = val.AsAny();
+            if (boxed != null && boxed.GetType() != targetType)
+            {
+                try
+                {
+                    boxed = Convert.ChangeType(boxed, targetType);
+                }
+                catch (System.InvalidCastException e)
+                {
+                    if (e.Message.Contains("RealmInteger"))
```

Comment:
This is added so that `RealmInteger` can be processed by the unit tests already in place.
Even though at line 577 in test `QueryFilter_WithAnyArguments_ShouldMatch` we just pass a RealmValue I still consider the test correct because the implicit conversion operator of RealmValue is already put at work during the creation of the test data (line 504-507).